packages/mobile/src/firebase/firebase.ts

```diff
@@ -26,57 +57,66 @@ export const initializeAuth = async (app: Firebase, address: string) => {
   Logger.info(TAG, 'Firebase Auth initialized successfully')
 }
 
-export const initializeCloudMessaging = async (app: Firebase, address: string) => {
-  // TODO(cmcewen): remove once we move off thunk
-  const store = getReduxStore()
-  const language = store.getState().app.language
-  const dispatch = store.dispatch
-
+export function* initializeCloudMessaging(app: Firebase, address: string) {
   Logger.info(TAG, 'Initializing Firebase Cloud Messaging')
-  const enabled = await app.messaging().hasPermission()
+
+  const enabled = yield call([app.messaging(), 'hasPermission'])
   if (!enabled) {
     try {
-      await app.messaging().requestPermission()
+      yield call([app.messaging(), 'requestPermission'])
     } catch (error) {
       Logger.error(TAG, 'User has rejected messaging permissions', error)
       throw error
     }
   }
 
-  const fcmToken = await app.messaging().getToken()
+  const fcmToken = yield call([app.messaging(), 'getToken'])
```

Comment:
Based on our conversation, consider adding a link to that saga issue in a comment here:
https://github.com/redux-saga/redux-saga/issues/27